utils/python/CIME/case.py

```diff
@@ -0,0 +1,80 @@
+"""
+Wrapper around all env XML for a case.
+
+All interaction with and between the module files in XML/ takes place
+through the Case module.
+"""
+
+from XML.standard_module_setup import *
+import CIME.utils
+from CIME.utils import expect, run_cmd
+from CIME.XML.machines import Machines
+
+from CIME.XML.env_test          import EnvTest
+from CIME.XML.env_mach_specific import EnvMachSpecific
+from CIME.XML.env_case          import EnvCase
+from CIME.XML.env_mach_pes      import EnvMachPes
+from CIME.XML.env_build         import EnvBuild
+from CIME.XML.env_run           import EnvRun
+from CIME.XML.env_archive       import EnvArchive
+from CIME.XML.env_batch         import EnvBatch
+
+class Case(object):
+
+    def __init__(self, case_root=os.getcwd()):
+        expect(os.path.isdir(case_root),
+               "Case root directory '%s' does not exist" % case_root)
+
+        self._env_files = []
+        self._env_files_that_need_rewrite = set()
+
+        self._env_files.append(EnvTest(case_root))
+        self._env_files.append(EnvMachSpecific(case_root))
+        self._env_files.append(EnvCase(case_root))
+        self._env_files.append(EnvMachPes(case_root))
+        self._env_files.append(EnvBuild(case_root))
+        self._env_files.append(EnvArchive(case_root))
+        self._env_files.append(EnvBatch(case_root))
+
+    def __del__(self):
+        self.flush()
+
+    def flush(self):
+        for env_file in self._env_files_that_need_rewrite:
+            env_file.write()
+
+        self._env_files_that_need_rewrite = set()
+
+    def get_value(self, item):
+        for env_file in self._env_files:
+            result = env_file.get_value(item)
+            if (result is not None):
+                return self.get_resolved_value(result)
```

Comment:
Hmm, it looks like the cross-XML dependency is an issue here as well. When I take out the get_resolved_value, I get this error:

OSError: [Errno 2] No such file or directory: '/home/jgfouca/acme/scratch/$CASE/bld'

Because EXEROOT doesn't get fully resolved.